packages/medusa/src/loaders/subscribers.ts

```diff
@@ -2,21 +2,24 @@ import glob from "glob"
 import path from "path"
 import { asFunction } from "awilix"
 import { MedusaContainer } from "../types/global"
+import { MedusaError } from "@medusajs/utils"
 
 /**
  * Registers all subscribers in the subscribers directory
  */
 export default ({ container }: { container: MedusaContainer }) => {
-  const isTest = process.env.NODE_ENV === "test"
-
-  const corePath = isTest
-    ? "../subscribers/__mocks__/*.js"
-    : "../subscribers/*.js"
+  const corePath = "../subscribers/*.js"
   const coreFull = path.join(__dirname, corePath)
 
   const core = glob.sync(coreFull, { cwd: __dirname })
   core.forEach((fn) => {
     const loaded = require(fn).default
+    if (!loaded) {
+      throw new MedusaError(
+        MedusaError.Types.UNEXPECTED_STATE,
+        `Subscriber ${fn} does not have a default export`
+      )
+    }
```

Comment:
suggestion: I wouldn't necessarely throw here but maybe warn that in this file there is no subscribers found. My thinking is that you could want to have a file that is utils for example here wdyt?

Reply:
Hmm I am not sure TBH, I thought it's better to "explode" early as it can otherwise be missed. Let's leave it like this for now as it is more explicit, WDYT @adrien2p ?